packages/apollo-graphql/src/transforms.ts

```diff
@@ -94,13 +95,37 @@ function sorted<T>(
   return undefined;
 }
 
+// Predicate for filtering and type checking operation definitions (query, mutation, subscription)
+// Similar to graphql's isExecutableDefinitionNode
+function isOperationDefinitionNode(
+  node: ASTNode,
+): node is OperationDefinitionNode {
+  return node.kind === 'OperationDefinition';
+}
+
+// Predicate for filtering and type checking fragment definitions
+// Similar to graphql's isExecutableDefinitionNode
+function isFragmentDefinitionNode(
+  node: ASTNode,
+): node is FragmentDefinitionNode {
+  return node.kind === 'FragmentDefinition';
+}
+
 // sortAST sorts most multi-child nodes alphabetically. Using this as part of
 // your signature calculation function may make it easier to tell the difference
 // between queries that are similar to each other, and if for some reason your
 // GraphQL client generates query strings with elements in nondeterministic
 // order, it can make sure the queries are treated as identical.
 export function sortAST(ast: DocumentNode): DocumentNode {
-  return visit(ast, {
+  const [operation] = ast.definitions.filter(isOperationDefinitionNode);
+  const fragments = ast.definitions.filter(isFragmentDefinitionNode);
```

Comment:
I fear that this is overly specific. Given the definition of [`DefinitionNode` and `ExecutableDefinitionNode`](https://github.com/graphql/graphql-js/blob/7dd2b2bed841b9fd0c40c328826363ec79cc65c4/src/language/ast.js#L201-L208):

```js
export type DefinitionNode =
  | ExecutableDefinitionNode
  | TypeSystemDefinitionNode
  | TypeSystemExtensionNode;

export type ExecutableDefinitionNode =
  | OperationDefinitionNode
  | FragmentDefinitionNode;
```

I don't know what `TypeSystemDefinitionNode` or `TypeSystemExtensionNode` are, but they are legal values in a `DefinitionNode`. This also would be susceptible to be other new things added in the future. With that lack of knowledge, I think it would make sense to favor using "fragments and everything else" versus "fragments and operations". Thoughts?

```js
const nonFragments = ast.definitions.filter((node) => ! isFragmentDefinitionNode(node))
const fragments = ast.definitions.filter(isFragmentDefinitionNode);
```

Reply:
I'm operating under the assumption that the document will only have `ExecutableDefinitionNodes` else it wouldn't be a valid operation signature. I wouldn't expect the other two `TypeSystemXNode`s to turn up in an operation (they're used in SDL) but that may be an incorrect assumption.

Regardless of what I find, I'll add some documentation here to explain.

Reply:
That is totally true for _our_ use case. I don't think you can be positive that it covers _all_ use cases for both us and consumers of `apollo-graphql`.

The function is called `sortAST` and accepts a `DocumentNode`; I think you need to account for all legal values that might exist in a `DocumentNode`.

Reply:
Left a piece off the above comment: this function is called `sortAST`; so it might be used to sort _any_ kid of AST. Restricting this to `DocumentNode`s that only include `ExecutableDefinitionNodes` doesn't feel right.

Reply:
I think giving this a more specific name, like `lexicographicSortOperations` (by analogy with [`lexicographicSortSchema`](https://github.com/graphql/graphql-js/blob/master/src/utilities/lexicographicSortSchema.js) might be more appropriate.

Reply:
Filtering out non-`ExecutableDefinitionNode`s (as you currently do) might be ok, that is also what `execute` does. Alternatively, maybe we could put them all at the end?

Reply:
Resolved in 2ad998b927ae2c1514514c536d65bfdd2b7b6e84

Following @martijnwalraven 's suggestion of parity with the graphql function, I've opted to rename (and effectively repurpose) this: `lexicographicSortOperations`.

This is worth discussing - it's a breaking change but this package is new and may not be in use. Regardless, this requires a major version bump.

Reply:
@abernix I'm just now seeing your comments as I had a stale page up while responding to comments. I'm open to not exporting this function publicly.